server/route/MessagingController.js

```diff
@@ -0,0 +1,224 @@
+	wss = new WebSocketServer({ server });
+
+	// WebSocket connection handling
+	wss.on("connection", (ws) => {
+		console.log("New WebSocket connection established");
+
+		// Handle incoming messages from clients
+		ws.on("message", async (message) => {
+			try {
+				const data = JSON.parse(message);
+				if (data.event === "joinRoom" && data.username) {
+					// Check if the username belongs to a valid account
+					const client = await Client.findOne({
+						username: data.username,
+					});
+					const stylist = await Stylist.findOne({
+						username: data.username,
+					});
+
+					if (!client && !stylist) {
+						ws.send(
+							JSON.stringify({
+								event: "error",
+								message:
+									"Invalid username: no matching account found",
+							})
+						);
+						return;
+					}
+
+					ws.username = data.username; // Attach the username to the WebSocket instance
+					connectedUsers[data.username] = ws; // Store the WebSocket connection
+					console.log(`${data.username} joined the room`);
+				} else if (data.event === "sendMessage") {
+					const { clientUsername, stylistUsername, sender, content } =
+						data;
+
+					// Validate data
+					if (
+						!clientUsername ||
+						!stylistUsername ||
+						!sender ||
+						!content
+					) {
+						console.error("Invalid message data");
+						ws.send(
+							JSON.stringify({
+								event: "error",
+								message: "Invalid message data",
+							})
+						);
+						return;
+					}
+
+					// Save the message to MongoDB
+					const newMessage = new Message({
+						clientUsername,
+						stylistUsername,
+						sender,
+						content,
+					});
+					await newMessage.save();
+
+					// Find the recipient WebSocket connection
+					const recipientUsername =
+						sender === clientUsername
+							? stylistUsername
+							: clientUsername;
+					const recipientWs = connectedUsers[recipientUsername];
+
+					if (
+						recipientWs &&
+						recipientWs.readyState === WebSocketServer.OPEN
+					) {
+						recipientWs.send(
+							JSON.stringify({
+								event: "receiveMessage",
+								clientUsername,
+								stylistUsername,
+								sender,
+								content,
+								createdAt: newMessage.createdAt,
+							})
+						);
+					}
+
+					console.log(
+						`Message from ${sender} sent to ${recipientUsername}`
+					);
+				}
+			} catch (error) {
+				console.error("Error handling message:", error);
+			}
+		});
+
+		ws.on("close", () => {
+			console.log("WebSocket connection closed");
+			if (ws.username) {
+				delete connectedUsers[ws.username]; // Remove the disconnected user
+			}
+		});
+	});
+};
+
+// Endpoint to fetch chat history between a client and a stylist
+router.get(
+	"/history",
+	asyncHandler(async (req, res, next) => {
+		const { clientUsername, stylistUsername } = req.query;
+
+		if (!clientUsername || !stylistUsername) {
+			return res
+				.status(400)
+				.json({ error: "Missing required query parameters" });
+		}
```

Comment:
Is it possible to use return(next(ConflictError("message")) to use the errorHandler like it is done for all other endpoints on the server?